src/Kdyby/Redis/RedisSessionHandler.php

```diff
@@ -85,11 +85,21 @@ protected function getTtl()
 		if ($this->ttl === NULL) {
 			if ($this->session !== NULL) {
 				$options = $this->session->getOptions();
-				$this->ttl = min($options['cookie_lifetime'], $options['gc_maxlifetime']);
+				if ($options['cookie_lifetime'] == 0) {
+					$this->ttl = $options['gc_maxlifetime'];
+				} else if ($options['gc_maxlifetime'] == 0) {
+					$this->ttl = $options['cookie_lifetime'];
+				} else {
+					$this->ttl = min($options['cookie_lifetime'], $options['gc_maxlifetime']);
+				}
 
 			} else {
 				$this->ttl = ini_get("session.gc_maxlifetime");
 			}
+
+			if (!$this->ttl) {
```

Comment:
This is not a good solution, beacause you can call `$handler->getTtl()`, catch the exception and it will keep working. It would be better to store the value in a local variable, validate it and only then set it.